runtime/lua/vim/treesitter/query.lua

```diff
@@ -3,14 +3,14 @@ local language = require('vim.treesitter.language')
 
 ---@class Query
 ---@field captures string[] List of captures used in query
----@field info TSQueryInfo Contains used queries, predicates, directives
+---@field info TSQueryInfo Contains information used in queries, predicates, directives
 ---@field query userdata Parsed query
 local Query = {}
 Query.__index = Query
 
 ---@class TSQueryInfo
 ---@field captures table
----@field patterns table<string,any[][]>
+---@field patterns table<integer,(integer|string)[][]> see |vim.treesitter.query.parse()|
```

Comment:
`(integer|string)[][][]` (see `query_inspect`)

Reply:
Not really; it might be correct if we say `(integer|string|nil)[][][]`, which is however confusing. The key (pattern_id) comes from the (1-based) index, in the order they appear in a query file, but they are not necessarily contiguous because some of them may not have predicates/directives. E.g.:

```lua
=vim.treesitter.query.get('c', 'locals').info
{
  captures = { "definition.function", "definition.macro", "scope", "definition.var", "definition.parameter", "definition.type", "definition.field", "definition", "reference" },
  patterns = {
    [15] = { { "set!", "reference.kind", "field" } },
    [16] = { { "set!", "reference.kind", "type" } }
  }
}
```
I will improve the documentation about the meaning of keys and values to be even clearer.

Reply:
Updated